daemon/cmd/daemon_main.go

```diff
@@ -1720,7 +1719,7 @@ func startDaemon(d *Daemon, restoredEndpoints *endpointRestoreState, cleaner *da
 			// the collection of stale AllowedIPs entries too early, leading to
 			// the disruption of otherwise valid long running connections.
 			if option.Config.KVStore != "" {
-				if err := ipcache.WaitForKVStoreSync(d.ctx); err != nil {
+				if err := params.IPIdentityWatcher.WaitForSync(d.ctx); err != nil {
```

Comment:
Random thought: We could also provide a identity watcher promise that resolves when the underlying watcher is ready/synced

Reply:
Yep, that could work indeed. It would be a bit tricky as of today, because we still need to have the object before to actually start the watcher, but that could be a follow-up once moving that to a start hook.